include/mbgl/map/camera.hpp

```diff
@@ -96,4 +98,23 @@ struct AnimationOptions {
         : duration(d) {}
 };
 
+/** Various options for describing properties of a physical camera hovering over a map.
+    All fields are optional.
+    */
+struct FreeCameraOptions {
+    /** Position of the camera in mercator coordinates */
+    optional<std::array<double, 3>> mercatorPosition;
+
+    /** Focus point on the map where the camera is looking at. Effectively the same value as
+        map center coordinate.
+    */
+    optional<LatLng> focusPoint;
+
+    /** Helper function for setting the mercator position with Lat&Lng and altitude */
+    void setLocation(const LatLng& location, double altitudeMeters);
```

Comment:
Should this take a tuple like getLocation returns?

Reply:
Would it be better to introduce a named struct for these values?

Reply:
@mpulkki-mapbox +1 for new struct https://github.com/mapbox/mapbox-gl-native/pull/16419#issuecomment-617155620